Source/Eto.Gtk/Forms/Controls/SplitterHandler.cs

```diff
@@ -11,37 +11,193 @@ public class SplitterHandler : GtkContainer<Gtk.Paned, Splitter, Splitter.ICallb
 		SplitterOrientation orientation;
 		SplitterFixedPanel fixedPanel;
 		int? position;
+		double relative = double.NaN;
+		int suppressSplitterMoved;
 
 		class EtoHPaned : Gtk.HPaned
 		{
+			public SplitterHandler Handler { get; set; }
+
 			#if GTK2
 			protected override void OnSizeRequested(ref Gtk.Requisition requisition)
 			{
-				base.OnSizeRequested(ref requisition);
-				if (PositionSet && Child1 != null)
+				var size1 = Child1.SizeRequest();
+				var size2 = Child2.SizeRequest();
+				if (Handler.position.HasValue)
+					size1.Width = Handler.position.Value;
+				else
+				{
+					var relative = Handler.relative;
+					var fixedPanel = Handler.fixedPanel;
+					if (!double.IsNaN(relative))
+					{
+						if (fixedPanel == SplitterFixedPanel.Panel1)
+							size1.Width = (int)Math.Round(relative);
+						else if (fixedPanel == SplitterFixedPanel.Panel2)
+							size2.Width = (int)Math.Round(relative);
+						else if (relative <= 0.0)
+							size1.Width = 0;
+						else if (relative >= 1.0)
+							size2.Width = 0;
+						else
+						{
+							// both get at least the preferred size
+							size1.Width = (int)Math.Round(Math.Max(
+								size1.Width/relative, size2.Width/(1-relative)));
+							size2.Width = 0;
+						}
+					}
+				}
+				requisition.Width = size1.Width + size2.Width + Handler.SplitterWidth;
+				requisition.Height = Math.Max(size1.Height, size2.Height);
+			}
+			#else
+			//TODO: Check GTK3 Sizing, this is just a guess!
+			protected override void OnGetPreferredWidth(out int minimum_width, out int natural_width)
+			{
+				int min1, width1, min2, width2, sw = Handler.SplitterWidth;
+				Child1.GetPreferredWidth(out min1, out width1);
+				Child2.GetPreferredWidth(out min2, out width2);
+				if (Handler.position.HasValue)
+				{
+					var pos = Handler.position.Value;
+					minimum_width = pos + sw + min2;
+					natural_width = pos + sw + width2;
+				}
+				else
 				{
-					var childreq = Child1.Requisition;
-					if (childreq.Width > 0)
-						requisition.Width += Position - childreq.Width;
+					var relative = Handler.relative;
+					var fixedPanel = Handler.fixedPanel;
+					if (!double.IsNaN(relative))
+					{
+						if (fixedPanel == SplitterFixedPanel.Panel1)
+							min1 = width1 = (int)Math.Round(relative);
```

Comment:
Looks like the `min` should not be altered by position and some code to adjust position to honor minimal sizes will be needed.